app/models/dish.rb

```diff
@@ -18,15 +21,12 @@ class Dish < ActiveRecord::Base
 
   MAX_DEBT = -80
 
-  def copy(new_user)
-    dish = Dish.find_by order: order, user: new_user
-    dish.delete if dish
-    new_dish = dup
-    new_dish.user = new_user
-    new_dish
-  end
-
   def subtract(shipping, payer)
-    user.subtract(price + shipping, payer)
+    users.each do |user|
+      user.subtract(price /
+                    user_dishes_count +
+                    shipping /
+                    order.user_appearances_in_order[user.id], payer)
```

Comment:
Why can't it be just `price + shipping / user_dishes_count`? What exactly does this method do? I can't imagine what `dish.substract(shipping, payer)` can possible do. 

Even if it's not your code, it's good to make it better if you do something with it.

Reply:
These `subtract`'s were really dirty and should be probably rewritten from 0. For now, the method in here divide price of a dish into `user_dishes_count` (which is count of users that share a dish - you pay for pizza 30$, it was divided between 3 sharers, so it only subtract 10$). 

Shipping is slightly non-intuitive - in here, `shipping`, which is already cost of shipping divided by `sum_users` (they are unique users in order), is divided by times specific user appears in order. Its needed for cases, when user is sharer of more than one dish - because while we calculate debts, each dish of an order is looping through every user that share it, we would subtract from such an user shipping cost more than once. To prevent that, it calculates how many times he appeared in order and divides shipping cost that many times